modules/snapshot/variables.tf

```diff
@@ -2,8 +2,9 @@ variable "lambda" {
   description = "Observe Lambda module"
   type = object({
     lambda_function = object({
-      arn  = string
-      role = string
+      arn           = string
+      role          = string
+      function_name = string
```

Comment:
This is correct but makes this PR a breaking change. While the variable claims it wants the Observe module the API contract is that it only wanted `arn` and `role` before.  Another approach that is non-breaking would extract this from the ARN:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/arn

```tf
resource "aws_lambda_invocation" "snapshot" {
  function_name = trimprefix(data.aws_arn.function.resource, "function:")
}
```

See https://github.com/observeinc/terraform-aws-lambda/blob/719216702da9216906b66da0f0cb2b31c3e950cf/modules/snapshot/main.tf#L60 for

[Optional attributes](https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes) exist for objects too and would also work here. But then you have to account for the unlikely case that `function_name` actually isn't passed and disable the resource via `count`.